packages/ros2/src/SystemHandle.hpp

```diff
@@ -81,7 +81,12 @@ class SystemHandle : public virtual FullSystem
 private:
 
   std::shared_ptr<rclcpp::Node> _node;
+#ifdef SOSS_ROS2__ROSIDL_GENERATOR_CPP
   std::unique_ptr<rclcpp::executor::Executor> _executor;
+#else
+  std::unique_ptr<rclcpp::Executor> _executor;
+#endif // SOSS_ROS2_ROSIDL_GENERATOR_CPP
```

Comment:
we should try to replace SOSS_ROS2_ROSIDL_GENERATOR_CPP with a USING_FOXY define instead.

Reply:
`USING_FOXY` wouldn't be completely accurate, since this macros is intended to be forward-compatible with future distributions. I also believe very strongly in "namespacing" macros (in this case, using `SOSS_ROS2__`) since macro pollution can become a serious headache. We could make the current more accurate by saying something like `SOSS_ROS2__USING_ROSIDL_GENERATOR_CPP`.

In this specific case, it's true that the change in namespace for the `Executor` class isn't actually related to the use of `rosidl_generator_cpp`, so a different macro name might be appropriate. The question in my mind, then, is if we should simply define a new macro under the same conditions as `SOSS_ROS2__USING_ROSIDL_GENERATOR_CPP` or if we should try to find a different sentinel that determines when `Executor` makes this breaking change to its API. I haven't thought of a way to do the latter, so unless there's a clever solution that I'm overlooking, I guess we should just piggyback off of the test for `SOSS_ROS2__USING_ROSIDL_GENERATOR_CPP`.